Assets/PlayerMovement.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerMovement : MonoBehaviour
+{
+
+    public CharacterController Controller;
+
+    [SerializeField]
+    public float MovementSpeed = 2f;
+    public float Speed = 12f;
+    public float Gravity = -9.81f;
+    public float JumpHeight = 3f;
+
+    public Transform GroundCheck;
+    public float GroundDistance = 4f;
+    public LayerMask GroundMask;
+
+    Vector3 velocity;
+    bool isGrounded;
+
+    void Start()
+    {
+        
+    }
+
+    void Update()
+    {
+        isGrounded = Physics.CheckSphere(GroundCheck.position, GroundDistance, GroundMask);
+
+        if(isGrounded && velocity.y < 0)
+        {
+            Debug.Log("Grounded");
+            velocity.y = -MovementSpeed;
+        }
+
+        float x = Input.GetAxis("Horizontal");
```

Comment:
Is this a distance or a delta for movement or something? It should be called something more descriptive if possible, like `xMovement`, `xDistance` etc